### ISSUE TITLE
InputNumber: Decimal Pad Zeroes

### ISSUE THREAD

**Issue by melloware** [primefaces/primefaces#6387]
This issue was reported again by a PRO User.
> If we input 19.950,000 in PF7 in the backing bean setter (server side) we receive a value (BigDecimal) with scale 3,
instead in PF8 we receive a value with scale 0.
> We need to have always receive a number with the same scale (in this case 3) also when decimal places are zeroes.

Exp;
on 7.x;
client: 19.950,000 server: 19950.000
on 8.x+
client: 19.950,000 server: 19950

See: https://github.com/primefaces/primefaces/pull/5505#issuecomment-578787754

**Comment by melloware** [comment]
Related to #5504

**Comment by metaxmx** [comment]
Oh, I didn't realise that additional decimal zeroes would influence the BigDecimal's scale property. Sorry I missed this in my original PR #5505 
Thanks for fixing it @melloware

**Comment by melloware** [comment]
No problem your fix was much appreciated!

**Comment by melloware** [comment]
For anyone that desires the 8.0 behavior you can simply apply this MonkeyPatch to your application
```javascript
if (PrimeFaces.widget.InputNumber) {
    PrimeFaces.widget.InputNumber.prototype.getValue = function() {
        return this.autonumeric.getNumericString();
    }
};
```

**Comment by melloware** [comment]
Circling back to this issue @mertsincan I wonder if we should only apply the padding if `padControl=true`??

```
padControl | true | Boolean | Controls padding of the decimal places. If true, always pads the decimal with zeros.
```

**Comment by melloware** [comment]
I created integration tests for this: https://github.com/primefaces-extensions/primefaces-integration-tests/commit/1f7f1d3ce505e736f3231310c44ac6ad1fb9916a

**Comment by metaxmx** [comment]
It seems like there is still a little Issue with the new `getValue()` function:

When I have a `<p:inputNumber decimalPlaces="0" />` with 0 decimal places (i.e. for an integer), the `getValue()` function still renders a decimal point (without any trailing decimals) like `"6."`.

Apparently the decimal places are provided not as number, but as a String in the `cfg`, so the check
```javascript
if (val && this.cfg.decimalPlaces) {
...
}
```
returns true, when `this.cfg.decimalPlaces == '0'` (because a String with "0" is truthy).

I believe, the check needs to either parse the number in the String `this.cfg.decimalPlaces` and compare it to `> 0` or needs to check explicitly against `this.cfg.decimalPlaces !== '0'`.

Do you want me to provide a more complex example, and a possibly a PR?

**Comment by melloware** [comment]
Hmm yes because my Integration Test using an integer is not showing any decimal places.

See this Integration Test: https://github.com/primefaces-extensions/primefaces-integration-tests/blob/master/src/test/java/org/primefaces/extensions/integrationtests/inputnumber/InputNumber001Test.java

**Comment by metaxmx** [comment]
> 
> 
> Hmm yes because my Integration Test using an integer is not showing any decimal places.
> 
> See this Integration Test: https://github.com/primefaces-extensions/primefaces-integration-tests/blob/master/src/test/java/org/primefaces/extensions/integrationtests/inputnumber/InputNumber001Test.java

Hm, strange, because I get NumberFormatExceptions in Selenium and verified it in the console [I mean my own example, not your integration test].
But maybe it's been fixed already. I'll check and get back to you ...

**Comment by melloware** [comment]
Let me know. Your fix does make sense just wondering why i am not seeing it.

**Comment by metaxmx** [comment]
I created an integration test to verify this issue:
https://github.com/metaxmx/primefaces-integration-tests/commit/237b14103985001ed726a4908d5ea5b7e4a386be

When you call `PrimeFaces.widgets.myInputNumber.getValue()` or look in the value of the Hidden Input, you'll find the "wrong" format (with a decimal point but without any decimal places). The new integration-test I linked above is currently therefore red.

However, when the value is sent to the server, it is parsed by a BigDecimal first, which seems to be fine with the given format. So you don't see any errors in the backend. Only, if you interact with the widget directly (or check the values of the hidden input via Selenium - which is what I'm doing).

Nevertheless, I think it should be fixed.
I think, the easies way would be something like this:
```javascript
// Instead of
if (val && this.cfg.decimalPlaces) { ...}

// something like:
if (val && parseInt(this.cfg.decimalPlaces, 10) > 0) { ... }

// e.g.
// parseInt(undefined, 10) > 0 === false
// parseInt('0', 10) > 0 === false
// parseInt('2', 10) > 0 === true
```

I've also create a pull request #6517 for this fix, if you agree with me
